lib/active_model/serializer/reflection.rb

```diff
@@ -119,6 +119,7 @@ def build_association(subject, parent_serializer_options)
               serializer_options(subject, parent_serializer_options, reflection_options)
             )
           rescue ActiveModel::Serializer::CollectionSerializer::NoSerializerError
+            warn "No serializer found for #{association_value}."
```

Comment:
this really needs to be a throw and catch as it's not exactly an exception, it's flow control.